In [2]:
import pandas as pd
import numpy as np


Importanto bando de datos de de url

In [3]:
url ='https://raw.githubusercontent.com/alura-cursos/pandas-conhecendo-a-biblioteca/main/base-de-dados/aluguel.csv'

In [4]:
df = pd.read_csv(url)
df

Tipo;Bairro;Quartos;Vagas;Suites;Area;Valor;Condominio;IPTU
0              Quitinete;Copacabana;1;0;0;40;1700;500;60         
1                  Casa;Jardim Botânico;2;0;1;100;7000;;         
2      Conjunto Comercial/Sala;Barra da Tijuca;0;4;0;...         
3                 Apartamento;Centro;1;0;0;15;800;390;20         
4             Apartamento;Higienópolis;1;0;0;48;800;230;         
...                                                  ...         
32955               Quitinete;Centro;0;0;0;27;800;350;25         
32956       Apartamento;Jacarepaguá;3;1;2;78;1800;800;40         
32957  Apartamento;São Francisco Xavier;2;1;0;48;1400...         
32958              Apartamento;Leblon;2;0;0;70;3000;760;         
32959  Conjunto Comercial/Sala;Centro;0;0;0;250;6500;...         

[32960 rows x 1 columns]

Al visualizar los datos percibimos que esta separado por ; por tanto vamos corregir la importación

In [7]:
df = pd.read_csv(url, sep=';')
df.head(5)

Tipo           Bairro  Quartos  Vagas  Suites  Area  \
0                Quitinete       Copacabana        1      0       0    40   
1                     Casa  Jardim Botânico        2      0       1   100   
2  Conjunto Comercial/Sala  Barra da Tijuca        0      4       0   150   
3              Apartamento           Centro        1      0       0    15   
4              Apartamento     Higienópolis        1      0       0    48   

    Valor  Condominio    IPTU  
0  1700.0       500.0    60.0  
1  7000.0         NaN     NaN  
2  5200.0      4020.0  1111.0  
3   800.0       390.0    20.0  
4   800.0       230.0     NaN

In [9]:
type(df)

pandas.core.frame.DataFrame

Para conocer la extrutura de nuestro data frame podemos usar el metodo shape



In [10]:
df.shape

(32960, 9)

Podemos observar que tiene 32960 filas y 9 columnas


Ahora vamos a evaluar las columnas que tenemos disponibles


In [17]:
df.columns

Index(['Tipo', 'Bairro', 'Quartos', 'Vagas', 'Suites', 'Area', 'Valor',
       'Condominio', 'IPTU'],
      dtype='object')

Ya sabemos los nombres de las columnas que conforman nuestro dataframe, ahora vamos a ver
que tipo de información esta contenida en ellas usando el metodo info

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32960 entries, 0 to 32959
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Tipo        32960 non-null  object 
 1   Bairro      32960 non-null  object 
 2   Quartos     32960 non-null  int64  
 3   Vagas       32960 non-null  int64  
 4   Suites      32960 non-null  int64  
 5   Area        32960 non-null  int64  
 6   Valor       32943 non-null  float64
 7   Condominio  28867 non-null  float64
 8   IPTU        22723 non-null  float64
dtypes: float64(3), int64(4), object(2)
memory usage: 2.3+ MB


In [22]:
df['Tipo']

0                      Quitinete
1                           Casa
2        Conjunto Comercial/Sala
3                    Apartamento
4                    Apartamento
                  ...           
32955                  Quitinete
32956                Apartamento
32957                Apartamento
32958                Apartamento
32959    Conjunto Comercial/Sala
Name: Tipo, Length: 32960, dtype: object

In [26]:
df[['Quartos','Valor']]

Quartos   Valor
0            1  1700.0
1            2  7000.0
2            0  5200.0
3            1   800.0
4            1   800.0
...        ...     ...
32955        0   800.0
32956        3  1800.0
32957        2  1400.0
32958        2  3000.0
32959        0  6500.0

[32960 rows x 2 columns]